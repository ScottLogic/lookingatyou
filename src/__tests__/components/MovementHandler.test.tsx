import { shallow } from 'enzyme';
import jsdom from 'jsdom';
import React from 'react';
import {
    MovementHandler,
    MovementHandlerProps,
} from '../../components/intelligentMovement/MovementHandler';

let props: MovementHandlerProps;
const imageData = { data: new Uint8ClampedArray(0), width: 10, height: 10 };
const mockUpdateAnimation = jest.fn();
const mockMathRandom = jest.fn();
const originalRandom = Math.random;

describe('Movement Handler', () => {
    beforeEach(() => {
        props = {
            width: 1920,
            height: 1080,
            environment: new jsdom.JSDOM(`...`, { url: 'http://localhost' })
                .window,
            fps: 1000,
            target: { x: 0, y: 0 },
            image: imageData,
            animation: [],
            updateAnimation: mockUpdateAnimation,
        };

        mockMathRandom.mockReturnValue(0.05);
        global.Math.random = mockMathRandom;
    });

    afterEach(() => {
        global.Math.random = originalRandom;
    });

    it('should render correctly', () => {
        const wrapper = shallow(<MovementHandler {...props} />).debug();
        expect(wrapper).toMatchSnapshot();
    });

    it('should dispatch updateAnimation when there are no detections', () => {
        jest.useFakeTimers();
        const wrapper = shallow(<MovementHandler {...props} />);
        wrapper.setProps({
            detections: [],
        });
        jest.advanceTimersByTime(20);
        expect(mockUpdateAnimation).toBeCalled();
    });
});
