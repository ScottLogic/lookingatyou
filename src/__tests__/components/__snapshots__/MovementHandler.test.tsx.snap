// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movement Handler should render correctly 1`] = `
"<div className=\\"movementHandler\\">
  <Connect(Component) dilation={1} detected={false} openCoefficient={0.45} isSleeping={false} width={1920} height={1080} environment={{...}} fps={1000} target={{...}} image={{...}} animation={{...}} animationExists={false} updateAnimation={[Function: mockConstructor]} />
  <Memo() text=\\"\\" show={false} />
</div>"
`;
